python/parquet_tools/csv2pq/schema.py

```diff
@@ -0,0 +1,300 @@
+        return ''
+    if pval1 < 0:
+        return ''
+    if len(pspec) == 1:
+        pval2 = 0
+    else:
+        try:
+            pval2 = int(pspec[1])
+        except Exception:  # We don't care what the exception is
+            return ''
+
+    # Do check for valid scale value
+    #
+    if pval2 < 0:
+        return ''
+
+    # Return the an int type if this has no digits after the decimal point
+    #
+    if pval2 == 0:
+        if pval1 < 3:
+            return 'int8'
+        elif pval1 < 5:
+            return 'int16'
+        elif pval1 < 10:
+            return 'int32'
+        else:
+            return 'int64'
+
+    # Return appropriate float
+    #
+    if pval1 < 7:
+        return 'float32'
+    return 'float64'
+
+
+# *****************************************************************************
+# *                              g e t _ t y p e                              *
+# *****************************************************************************
+
+def get_type(cname, ctype, cnull, colnulls):
+    """Record correct type for a particular column.
+    In:  cname    - The name of the column.
+         ctype    - The proposed column data type.
+         cnull    - Boolean: True if column may be null.
+         colnulls - A list to which null column info is to be appended.
+    Out: The [column name, type name, and pandas datatype] is appended
+         to the colSpecs and pandas type is is entered into he ColTypes
+         dictionary using the column name as the key both in the CmdInfo class.
+         If the column may be null, the [column number, cname_ISNULL] is
+         appended to the colnulls list passed as an argument. The type for
+         this additional column is also added to the column type dictionary.
+         This function returns None on success and exits upon error.
+    """
+
+    # We check for types using lower case but use original spec in messages
+    #
+    xtype = ctype.lower()
+
+    # Check for character type and decimal type
+    #
+    if xtype[0:4] == 'char':
+        xtype = 'char'
+    elif xtype[0:8] == 'decimal(':
+        xtype = get_dectype(xtype[8:])
+
+    # All remaining types must be found in our name to numpy dictionary
+    #
+    CmdInfo.colSpecs.append([cname, ctype, xtype])
+    if xtype in TypeInfo.table:
+        CmdInfo.colTypes[cname] = TypeInfo.table[xtype]
+    else:
+        fatal(4, cname, ctype)
+
+    # New record whether we should preprocess null values for the column
+    #
+    if cnull and xtype in TypeInfo.isInt:
+        cname += '_ISNULL'
+        colnulls.append([len(CmdInfo.colNames)-1, cname])
+        CmdInfo.colTypes[cname] = np.bool_
+
+
+# *****************************************************************************
+# *                            g e t _ s c h e m a                            *
+# *****************************************************************************
+
+def get_schema(schfile, do_display):
```

Comment:
I'd probably make a separate class which includes all methods in this modules, and also moved `CmdInfo.colTypes`, `CmdInfo.colSpecs`, etc. in this class to so that you have all schema-related stuff in one place.